### ISSUE TITLE
genai: In NewClient() c.SetGoogleClientInfo undefined (type *aiplatform.PredictionClient has no field or method SetGoogleClientInfo)

### ISSUE THREAD

**Issue by codecrane** [googleapis/google-cloud-go#9167]
**Code:**

https://github.com/googleapis/google-cloud-go/blame/77c1cfac704dab6704e9b55bdd8a710695dc936a/vertexai/genai/client.go#L56

```
func NewClient(ctx context.Context, projectID, location string, opts ...option.ClientOption) (*Client, error) {
	opts = append([]option.ClientOption{
		option.WithEndpoint(fmt.Sprintf("%s-aiplatform.googleapis.com:443", location)),
	}, opts...)
	c, err := aiplatform.NewPredictionClient(ctx, opts...)
	if err != nil {
		return nil, err
	}
	c.SetGoogleClientInfo("gccl", internal.Version)
	return &Client{
		c:         c,
		projectID: projectID,
		location:  location,
	}, nil
}
```

**Implication:**
Application does not compile as SetGoogleClientInfo() is not available in the pinned version of the Prediction client. It looks like NewClient() is attempting to call the prediction client internal `c.setGoogleClientInfo()`
https://github.com/googleapis/google-cloud-go/blob/main/aiplatform/apiv1beta1/prediction_client.go#L412 

```
# cloud.google.com/go/vertexai/genai
../../go/pkg/mod/cloud.google.com/go/vertexai@v0.5.1/genai/client.go:56:4: c.SetGoogleClientInfo undefined (type *aiplatform.PredictionClient has no field or method SetGoogleClientInfo)
```

**Solution:**
Remove the call to `c.SetGoogleClientInfo("gccl", internal.Version)` from `func NewClient(ctx context.Context, projectID, location string, opts ...option.ClientOption) (*Client, error) {`

**Comment by codyoss** [comment]
Will get a patch out but as a workaround you can pull cloud.google.com/go/aiplatform@v1.58.0, sorry about that

**Comment by codyoss** [comment]
Thank you for the report! Released in 0.5.2
